blocks/logic.ts

```diff
@@ -605,43 +626,43 @@ const LOGIC_COMPARE_ONCHANGE_MIXIN = {
 /**
  * "logic_compare" extension function. Adds type left and right side type
  * checking to "logic_compare" blocks.
- * @this {Block}
- * @readonly
  */
-const LOGIC_COMPARE_EXTENSION = function() {
+const LOGIC_COMPARE_EXTENSION = function(this: CompareBlock) {
   // Add onchange handler to ensure types are compatible.
   this.mixin(LOGIC_COMPARE_ONCHANGE_MIXIN);
 };
```

Comment:
Unrelated to types but: is this done to work around issues with mixing in methods that override built-in methods? We might want to add a comment documenting that.

Reply:
There's no problem overriding built-in methods with a JS-style block definition (i.e., a mixin that gets its properties copied on the the `Block` instance).  But the block that uses the `LOGIC_COMPARE_ONCHANGE_MIXIN` is defined using JSON, and I gather that the way we provided to mix arbitrary extra properties onto JSON-defined blocks is to use an extension like this.  (Which is probably sensible, since the extension can do all kinds of other things to the block, too.)